### DISCUSSION TITLE
custom URL

### DISCUSSION THREAD

**Discussion by mfrigerio17** [abraunegg/onedrive#2304] [Q&A] [answered]
Hi, I have access to a SharePoint shared drive from my previous company.
I don't have a Microsoft account, but when I go to the URL of that shared drive, I input my email address (which is recognized, as opposed to when I go to a standard login page), and then I receive a code in my email for the authentication.

Now, how can I customize the URL that the onedrive client uses (the one that it prompts the user with, at the initial run), with the one for my company?

Thank you

**Comment by abraunegg** [comment, marked as answer]
@mfrigerio17 
> Now, how can I customize the URL that the onedrive client uses (the one that it prompts the user with, at the initial run), with the one for my company?

All authentication needs to be done against `login.microsoftonline.com` as this is the OneDrive API requirement. 

You can try adding the Azure Tenant ID to your config file:
```
azure_tenant_id = "my.custom.url"
```

When you attempt to authenticate, this is the URL it then will ask you to visit:
```
Configuration file successfully loaded
Configuring Global Azure AD Endpoints - Single Tenant Application
Authorize this app visiting:

https://login.microsoftonline.com/my.custom.url/oauth2/v2.0/authorize?client_id=d50ca740-c83f-4d1b-b616-12c519384f0c&scope=Files.ReadWrite%20Files.ReadWrite.All%20Sites.ReadWrite.All%20offline_access&response_type=code&prompt=login&redirect_uri=https://login.microsoftonline.com/my.custom.url/oauth2/nativeclient

Enter the response uri: 
```

The default URL is:
```
Configuring Global Azure AD Endpoints
Authorize this app visiting:

https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id=d50ca740-c83f-4d1b-b616-12c519384f0c&scope=Files.ReadWrite%20Files.ReadWrite.All%20Sites.ReadWrite.All%20offline_access&response_type=code&prompt=login&redirect_uri=https://login.microsoftonline.com/common/oauth2/nativeclient

Enter the response uri: 
```

If this does not work for you - then there is no solution for you due to how the OneDrive API works.

**Reply by mfrigerio17** [reply]
Hi, thanks for your answer, the `azure_tenant_id` actually worked, I managed to land in the appropriate login page.
However, after putting my code, it does not login and I get redirected to an error page, mentioning

AADSTS900971: No reply address provided.

Not sure whether this is solvable, and whether it may be useful for you to investigate, to improve the app

**Reply by abraunegg** [reply]
There is nothing here to investigate from my side.

The application, when using 'azure_tenant_id' configures the correct response URL that is to be provided on successful login.

When this issue occurs, it is typically when you have configured your own application ID to use with the application and you have not configured this correctly.

Refer to: https://answers.microsoft.com/en-us/msteams/forum/all/aadsts900971-no-reply-address-provided/1967cfa2-fde1-4359-b66d-bc5b0d7dacfe
